src/services/public_http_server/handlers/mark_all_as_read.rs

```diff
@@ -0,0 +1,57 @@
+use {
+    crate::{
+        error::NotifyServerError,
+        model::helpers::{get_project_by_project_id, mark_all_notifications_as_read_for_project},
+        rate_limit::{self, Clock, RateLimitError},
+        registry::{extractor::AuthedProjectId, storage::redis::Redis},
+        state::AppState,
+    },
+    axum::{
+        extract::State,
+        http::StatusCode,
+        response::{IntoResponse, Response},
+    },
+    relay_rpc::domain::ProjectId,
+    std::sync::Arc,
+    tracing::instrument,
+};
+
+#[instrument(name = "mark_all_as_read", skip(state))]
+pub async fn handler(
+    State(state): State<Arc<AppState>>,
+    AuthedProjectId(project_id, _): AuthedProjectId,
+) -> Result<Response, NotifyServerError> {
+    if let Some(redis) = state.redis.as_ref() {
+        mark_all_as_read_rate_limit(redis, &project_id, &state.clock).await?;
+    }
+
+    let project = get_project_by_project_id(project_id, &state.postgres, state.metrics.as_ref())
+        .await
+        .map_err(|e| match e {
+            sqlx::Error::RowNotFound => {
+                NotifyServerError::UnprocessableEntity("Project not found".into())
+            }
+            e => e.into(),
+        })?;
+
+    mark_all_notifications_as_read_for_project(project.id, &state.postgres, state.metrics.as_ref())
+        .await?;
+
+    Ok(StatusCode::NO_CONTENT.into_response())
+}
+
+pub async fn mark_all_as_read_rate_limit(
```

Comment:
Is there any reason to make this single call in a function? Since it's just used once and just called another function inside?

Reply:
It's just a pattern I've been doing when copying endpoints. Sometimes I test the rate limit. It's nice to have the config bundled here so you can see clearly what parameters are involved. No strong reason though

Reply:
I think it's also nice that it keeps biz logic separate from auxiliary processes and gives you a tidy view of what the main function is supposed to do.

Reply:
Just let's keep it simple if we can 🙌